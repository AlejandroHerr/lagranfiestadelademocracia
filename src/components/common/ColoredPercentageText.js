import React from 'react';
import PropTypes from 'prop-types';

import ColoredNumber from './ColoredNumber';
import PercentageText from './PercentageText';

const ColoredPercentageText = ({ children, ...props }) => (
  <ColoredNumber renderComponent={PercentageText} {...props}>
    {children}
  </ColoredNumber>
);

ColoredPercentageText.propTypes = {
  children: PropTypes.number.isRequired,
};

export default ColoredPercentageText;
