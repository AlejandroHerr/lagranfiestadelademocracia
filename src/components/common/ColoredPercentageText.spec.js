import React from 'react';
import renderer from 'react-test-renderer';

import ColoredPercentageText from './ColoredPercentageText';

describe('components/component/ColoredPercentageText', () => {
  it('renders correctly for positive numbers', () => {
    const tree = renderer.create(<ColoredPercentageText>{1}</ColoredPercentageText>).toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('renders correctly for negative numbers', () => {
    const tree = renderer.create(<ColoredPercentageText>{-0.001}</ColoredPercentageText>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
