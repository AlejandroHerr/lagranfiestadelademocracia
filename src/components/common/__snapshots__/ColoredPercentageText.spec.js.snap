// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/component/ColoredPercentageText renders correctly for negative numbers 1`] = `
<div
  className="sc-bdVaJa sc-htpNat chjhjd"
  color="red"
  fontFamily="body"
>
  -0.1%
</div>
`;

exports[`components/component/ColoredPercentageText renders correctly for positive numbers 1`] = `
<div
  className="sc-bdVaJa sc-htpNat ijHaiK"
  color="normal"
  fontFamily="body"
>
  100%
</div>
`;
